File: projects/de.genesez.platforms.php.umlsupport/src-gen/UML/Association.php
<?php

/* PROTECTED REGION ID(php.own.imports._16_0_b6f02e1_1239222572640_136215_385) ENABLED START */
// TODO: put your further include + require statements here
/* PROTECTED REGION END */

/**
 * principal type for all associations, defines a universal interface to deal 
 * with all kinds of associations.
 * @author	dreamer
 * @package	UmlSupport
 */
interface UML_Association  {
	
	
	
	/**
	 * inserts an associated object into the association, if not already present.
	 * @param	mixed	$associated	the object to create a link with
	 * @return	mixed
	 */
	public function insert($associated);

	/**
	 * removes an associated object from the association, if present
	 * @param	mixed	$associated	
	 * @return	mixed
	 */
	public function remove($associated);

	/**
	 * in case of a to-one-association, return the associated object,<br>otherwise 
	 * return the first associated object and subsequent calls iterate over the 
	 * associated objects
	 * @return	mixed
	 */
	public function get();

	/**
	 * returns all associated objects as an associated array as a copy of the 
	 * internal data structure
	 * @return	array of mixed
	 */
	public function getAll();

	/**
	 * returns an
	 * @return	Iterator
	 */
	public function iterator();

	
	
	// -- own code declarations ---------------------------------------------
	/* PROTECTED REGION ID(php.interface.own.code.declaration._16_0_b6f02e1_1239222572640_136215_385) ENABLED START */
	// TODO: put your further code declarations for interface 'UML_Association' here
	/* PROTECTED REGION END */
}
?>

File: projects/de.genesez.platforms.java.umlsupport/src/org/genesez/platform/java/umlsupport/associations/Association.java
package org.genesez.platform.java.umlsupport.associations;

import java.util.Collection;
import java.util.Iterator;

/**
 * principal type for all associations, defines a universal interface to deal
 * with all kinds of associations.
 * 
 * @author georg beier
 * 
 * @param <From>
 *            the from class
 * @param <To>
 *            the to class
 */
public interface Association<From, To> extends Iterable<To> {
	/**
	 * insert associated object into the association, if not already present.
	 * 
	 * @param associated
	 *            the newly associated object
	 * @return associated
	 */
	To insert(To associated);

	/**
	 * remove an associated object from the association, if present
	 * 
	 * @param associated
	 *            object to be removed
	 * @return the actually removed object or null
	 */
	To remove(To associated);

	/**
	 * in case of a to-one-association, return the associated object, otherwise
	 * return the first object that would be returned by iterator().next().
	 * 
	 * @return an (or the) associated object or null, if none exists
	 */
	To get();
	
	/**
	 * returns all associated objects as an unmodifiable collection
	 * 
	 * @return the associated objects or an empty collection
	 */
	Collection<To> getAll();
	
	/**
	 * get an iterator on all associated objects
	 * 
	 * @see java.lang.Iterable#iterator()
	 * @return an iterator on the associated objects, even if it is a to-one
	 *         association
	 */
	Iterator<To> iterator();
}
